### DISCUSSION TITLE
Add a recommended price for minimum price items

### DISCUSSION THREAD

**Discussion by BaderSZ** [btcpayserver/btcpayserver#3874] [Ideas / Feature Requests]
One useful feature when using minimum prices in the point of sale app is to show a suggested value instead of the minimum. This is commonly done for co-ops, and useful for donations, e.g. set the minimum donation to 1 USD but a recommended donation to 5 USD.

I've been [working on this feature](https://github.com/BaderSZ/btcpayserver/tree/minimum-with-suggestion) (of course with mediocre C# code, since I only started using this language) and would like feedback, especially design suggestions. So far:

<p align="center">
  <p>Edit (disabled)</p>
  <img src="https://user-images.githubusercontent.com/2165320/173859354-ac9549f5-10ba-4472-83fd-6a2354a91bb1.png">
 <p> Edit (enabled)</p>
  <img src="https://user-images.githubusercontent.com/2165320/173859284-a2225d1f-fa09-417b-a7bc-2119ac04ce3e.png">
<p>Raw Editor</p>
  <img src="https://user-images.githubusercontent.com/2165320/173859395-c90d41d1-3250-4345-8a5e-f37902fba3d5.png">
<p>Item Shop</p>
  <img src="https://user-images.githubusercontent.com/2165320/173859424-f2284c1e-d33b-46f9-be66-e43980e2cb1d.png">
<p>Print</p>
  <img src="https://user-images.githubusercontent.com/2165320/173859459-6a6080db-b858-4d44-a386-3a9060e61e7a.png">
</p>

Any suggestions?

**Comment by pavlenex** [comment]
I don't understand what price recommendation does and why it is needed as  core functionality of the point of sale app?

I've never seen a point of sale terminal with such feature. Please understand that point of sale is not meant for donations, and any edge-case feature shouldn't be added just for the sake of adding it.

That said I appreciate your efforts in coding and showcasing it, I just don't see how it's useful for a point of sale app. Consider making it a plugin rather than a core feature.

**Comment by Kukks** [comment]
I think he labeled it wrong which gives off the wrong impression. The
correct term is "default price", which applies only for the minimum price
type.

**Reply by BaderSZ** [reply]
> I think he labeled it wrong which gives off the wrong impression. The correct term is "default price", which applies only for the minimum price type.

Could be. That is how I thought of it as this was our use-case. A default price would be more general.

If you believe a plugin is still fitting, I could close the discussion and publish it seperately. :)

**Reply by bolatovumar** [reply]
@BaderSZ wouldn't it be simpler to create a separate price type? This way you can have a "Minimum" price type which enforces the minimum price and you can also have a "Recommended" price type which is like the "Minimum" one but it doesn't enforce the minimum type, it just sets the default price and user can still lower it.

**Reply by BaderSZ** [reply]
That could also be done, if you wish. It really is your decision if there's an interest for an MR.

**Comment by Kukks** [comment]
Actually, the minimum price thing is already implemented in the code. If
the item is priced at 10, you can send an http post with a price of 11 and
it will be valid

**Reply by BaderSZ** [reply]
Yes, it is. The issue is mostly in PoS app. I could've created a html form on my site, but I couldn't select or modify what the default value shown was. Or am I missing something?
